### PR TITLE
[Tracing] Custom tracing collector and querier.

### PR DESCRIPTION
**Pull request by robin-near**
See README.md

### CODE REVIEW

File: tracing/src/db.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/// Simple wrapper on top of mongodb.
+#[derive(Clone)]
+pub struct Database {
+    pub db: mongodb::Database,
+}
+
+impl Database {
+    /// The collection where we dump all the traces received by the collector.
+    pub fn span_chunks(&self) -> Collection<SpanChunk> {
+        self.db.collection("SpanChunk")
```
Comment (tayfunelmas) [inline tracing/src/db.rs:14]: does "Chunk" refer to our block chunk or a db concept here? if the former, can we use another name, as the trace will not be only chunks in some cases.

File: tracing/src/primitives.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+use bson::oid::ObjectId;
+use bson::Binary;
+use serde::{Deserialize, Serialize};
+
+// The storage format in MongoDB.
+#[derive(Serialize, Deserialize)]
+pub struct SpanChunk {
```
Comment (tayfunelmas) [inline tracing/src/primitives.rs:7]: maybe not use "Chunk" here to not confuse with our concept of block chunks?

File: tracing/src/queries.rs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+) -> Result<HttpResponse, Error> {
+    // TODO: Enable compression as the data is very compressible
+    // TODO: Set a limit on the duration of the request interval.
+    let col = data.db.span_chunks();
+    let mut chunks = col
+        .find(
+            doc! {
+                "max_time": {"$gt": req.start_timestamp_unix_ms * 1000000},
+                "min_time": {"$lt": req.end_timestamp_unix_ms * 1000000},
+            },
```
Comment (tayfunelmas) [inline tracing/src/queries.rs:109]: to confirm, here we are checking if there any intersection with the spanchunk right? and in this case, we still return the all the spans/events even if they may not be part of the intersection?

File: tracing/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  mongo-express:
+    image: mongo-express
+    restart: always
+    ports:
+      - 8081:8081
+    environment:
+      ME_CONFIG_MONGODB_ADMINUSERNAME: root
+      ME_CONFIG_MONGODB_ADMINPASSWORD: example
+      ME_CONFIG_MONGODB_URL: mongodb://root:insecure@mongo:27017/
+      ME_CONFIG_BASICAUTH: false
```
Comment (tayfunelmas) [inline tracing/docker-compose.yml:20]: the docker compose on my machine complained about this and I had to change it to "false" (with quotes)

File: tracing/src/db.rs (hunk 1 above)
Comment (robin-near) [reply tracing/src/db.rs:14]: It's just a db concept meaning a chunk of spans.

File: tracing/src/primitives.rs (hunk 2 above)
Comment (robin-near) [reply tracing/src/primitives.rs:7]: Yeah I guess RawTrace is better.

File: tracing/src/queries.rs (hunk 3 above)
Comment (robin-near) [reply tracing/src/queries.rs:109]: Yep. We can add better filtering later.

File: tracing/docker-compose.yml (hunk 4 above)
Comment (robin-near) [reply tracing/docker-compose.yml]: Added quotes.

**Review by tayfunelmas — APPROVED** [review]

**Pull request merged**